samples/python/agents/signing_and_verifying/__main__.py

```diff
@@ -0,0 +1,122 @@
+import json
+from pathlib import Path
+
+import uvicorn
+from a2a.server.apps import A2AStarletteApplication
+from a2a.server.request_handlers import DefaultRequestHandler
+from a2a.server.tasks import InMemoryTaskStore
+from a2a.types import (
+    AgentCapabilities,
+    AgentCard,
+    AgentSkill,
+)
+from a2a.utils.signing import create_agent_card_signer
+from cryptography.hazmat.primitives import asymmetric, serialization
+from starlette.responses import FileResponse
+from starlette.routing import Route
+
+from agent_executor import (
+    SignedAgentExecutor,
+)
+
+if __name__ == "__main__":
+    # Generate a private, public key pair
+    private_key = asymmetric.ec.generate_private_key(asymmetric.ec.SECP256R1())
+    public_key = private_key.public_key()
+
+    # Save public key to a file
+    pem = public_key.public_bytes(
+        encoding=serialization.Encoding.PEM,
+        format=serialization.PublicFormat.SubjectPublicKeyInfo,
+    ).decode("utf-8")
+    kid = "my-key"
+    keys = {}
+    try:
+        with Path("public_keys.json").open() as f:
+            keys = json.load(f)
+    except FileNotFoundError:
+        pass
+    except json.JSONDecodeError:
+        print("Warning: public_keys.json is not valid JSON. Starting fresh.")
+
+    keys[kid] = pem
+
+    with Path("public_keys.json").open("w") as f:
```

Comment:
nit: maybe adding public_keys.json in this folder to .gitignore to avoid committing it accidentally?